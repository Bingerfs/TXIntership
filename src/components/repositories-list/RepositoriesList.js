import React, { Component } from 'react';
import { Card, CardImg, CardTitle, CardBody, CardText } from 'reactstrap';

class RepositoriesList extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const userCards = this.props.users.map((user) => {
            const profileAvatar = user.avatar_url || 'assets/images/Generic-Profile-1600x1600.png'; 
            return (
                <div key='user.nickName' className="col-12 m-1">
                    <Card>
                        <div className='row'>
                            <div className='col-3'>
                                <CardImg src={ profileAvatar } alt={ user.nickName } />
                            </div>
                            <div className='col'>
                            <CardBody>
                            <CardTitle tag='h2'>{ user.nickName }</CardTitle>
                            <CardText>
                                Full Name: { user.name }
                                <br></br>
                                Github repo: <a href={ user.github_url }>{ user.github_url }</a>
                            </CardText>
                        </CardBody>
                            </div>
                        </div>
                    </Card>
                </div>
            );
        });
        return (
            <div className='container'>
                <div className='row'>
                { userCards }
                </div>
            </div>
        );
    }
}

export default RepositoriesList;