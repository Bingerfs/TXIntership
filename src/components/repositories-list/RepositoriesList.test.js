import react from 'react';
import { shallow } from 'enzyme';
import RepositoriesList from './RepositoriesList.js';

describe('RepositoriesList', () => {
    it("should render correctly", () => {
        const users = [];
        const component = shallow(<RepositoriesList users={ users } />);
      });

    it("should render initial layout", () => {
        const users = [];
        const component = shallow(<RepositoriesList users={ users }/>);
        expect(component.getElements()).toMatchSnapshot();
    });
})