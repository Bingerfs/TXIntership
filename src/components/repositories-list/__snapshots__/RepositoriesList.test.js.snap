// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesList should render initial layout 1`] = `
Array [
  <div
    className="container"
  >
    <div
      className="row"
    />
  </div>,
]
`;
